core/decorators.py

```diff
@@ -0,0 +1,177 @@
+from viur.core import errors, current
+from typing import Callable, Dict, Union, List
+
+import functools
+import logging
+import types
+import warnings
+
+
+def ensure_viur_flags(func: Callable) -> None:
+    try:
+        func.viur_flags
+    except AttributeError:
+        func.viur_flags = {}
+
+
+def access(*access: str | list[str]) -> Callable:
+    """Decorator, which performs the authentication and authorization check.
+
+    To expose a method only to logged in users with the access
+    "root" or ("admin" and "file-edit") or "maintainer"
+    use this decorator like this:
+
+    .. code-block:: python
+        from viur.core.decorators import access
+        @access("root", ["admin", "file-edit"], ["maintainer"])
+        def yourMethod(self):
```

Comment:
You missed the `snake_case` suggestion in https://github.com/viur-framework/viur-core/pull/782#discussion_r1256163015

```suggestion
        def your_method(self):
```